semgrep/semgrep/rule.py

```diff
@@ -70,213 +52,61 @@ def __init__(self, raw: YamlTree[YamlMap]) -> None:
         # add typescript to languages if the rule supports javascript.
         if any(language == Language.JAVASCRIPT for language in rule_languages):
             rule_languages.add(Language.TYPESCRIPT)
+            self._raw["languages"] = [r.value for r in rule_languages]
 
         self._languages = sorted(rule_languages, key=lambda lang: lang.value)  # type: ignore
 
         # check taint/search mode
         if self._raw.get("mode") == JOIN_MODE:
             self._mode = JOIN_MODE
         else:
-            self._expression, self._mode = self._build_search_patterns_for_mode(
-                self._yaml
-            )
+            self._mode = SEARCH_MODE
 
         if any(language == Language.REGEX for language in self._languages):
             self._validate_none_language_rule()
 
-    def __eq__(self, other: object) -> bool:
-        if not isinstance(other, type(self)):
-            return False
-
-        return self._raw == other._raw
-
-    def __hash__(self) -> int:
-        return hash(self.id)
-
-    def has_pattern_where_python(self) -> bool:
+    def _validate_none_language_rule(self) -> None:
```

Comment:
Rewrote the validate_none_language_rule to not need to parse into BooleanExpression. The check before is equivalent to making sure `pattern` does not exist in a regex rule.